doc/whats_new/upcoming_changes/sklearn.utils/29818.api.rst

```diff
@@ -1,4 +1,7 @@
-- :func:`check_estimators.check_sample_weights_invariance` replaced by
-  :func:`check_estimators.check_sample_weight_equivalence` which uses
-  integer (including zero) weights.
+- :func:`utils.estimator_checks.check_sample_weights_invariance`
+  replaced by
+  :func:`check_sample_weight_equivalence_on_dense_data`
```

Comment:
It would be more like

```suggestion
  :func:`~utils.estimator_checks.check_sample_weight_equivalence_on_dense_data`
```

but we don't really need to annotate them (or even have a changelog for them) since they're not really public and don't get rendered in the API pages. So this would only raise a warning in sphinx build.

That said, I'm in favor of adding nice docstrings to tests and adding them to the API pages (not in this PR). However, this changelog needs to remove the `:func:` directives, or the mentions of them completely.

Reply:
If I understood correctly the sphinx doc, the tilde suppresses the cross-reference, and I should therefore add it for all the functions of this PR as they are not rendered on the API pages ?

Reply:
The tilde only changes the rendered version from the full qual path to only rendering the name of the function, but still trying to resolve the hyperlink, which raises a warning in this case.

Reply:
Thanks for the clarification ! I remove the `:func:` directives completely.